docs/ExceptionHandling.md

```diff
@@ -0,0 +1,60 @@
+# Exception handling in tests
```

Comment:
Whichever way we go, we probably ought to describe the limitations of what we have and the recommended route for exception expectation.